### PR TITLE
update: Added decide API documentation to the readme and removed the old APIs from it

### PR DESCRIPTION
**Pull request by zashraf1985**
## Summary
Added decide API documentation to the read me and removed the old Experiment and Feature components and hooks from it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,15 +24,24 @@ The React SDK is compatible with `React 16.3.0 +`
...
   createInstance,
   OptimizelyProvider,
-  OptimizelyExperiment,
-  OptimizelyVariation,
-  OptimizelyFeature,
+  useDecision,
 } from '@optimizely/react-sdk';
 
-const optimizely = createInstance({
+const optimizelyClient = createInstance({
```
Comment (yavorona) [inline README.md:30]: Should also be updated on line 49

File: README.md [hunk 2]
```diff
@@ -108,13 +97,13 @@ const optimizely = createInstance({
...
 
-Required at the root level. Leverages React’s `Context` API to allow access to the `ReactSDKClient` to components like `<OptimizelyFeature>` and `<OptimizelyExperiment>`.
+Required at the root level. Leverages React’s `Context` API to allow access to the `ReactSDKClient` to the `useDecision` hook.
 
 _props_
 
 - `optimizely : ReactSDKClient` created from `createInstance`
 - `user: { id: string; attributes?: { [key: string]: any } } | Promise` User info object - `id` and `attributes` will be passed to the SDK for every feature flag, A/B test, or `track` call, or a `Promise` for the same kind of object
-- `timeout : Number` (optional) The amount of time for OptimizelyExperiment and OptimizelyFeature components to render `null` while waiting for the SDK instance to become ready, before resolving..
+- `timeout : Number` (optional) The amount of time for `useDecision` to return `null` flag Decision while waiting for the SDK instance to become ready, before resolving..
```
Comment (yavorona) [inline README.md:106]: nit. Extra dot at the end

File: README.md [hunk 3]
```diff
@@ -530,13 +378,23 @@ import * as ReactDOMServer from 'react-dom/server';
...
+      { !decision.enabled && <p>The feature is not enabled</p> }
+      { decision.variationKey === 'variation1' && <p>Variation 1</p> }
+      { decision.variationKey === 'variation2' && <p>Variation 2</p> }
+    </React.Fragment>
+  );
+}
+
 async function main() {
   const resp = await fetch('https://cdn.optimizely.com/datafiles/<Your-SDK-Key>.json');
   const datafile = await resp.json();
```
Comment (yavorona) [inline README.md:400]: Line below: We can call it `optimizelyClient` as in other code samples.

File: README.md [hunk 4]
```diff
@@ -546,18 +404,7 @@ async function main() {
 
   const output = ReactDOMServer.renderToString(
     <OptimizelyProvider optimizely={optimizely} user={{ id: 'user1' }} isServerSide={true}>
```
Comment (yavorona) [inline README.md:406]: should be `{optimizelyClient}` as in other code samples.

File: README.md [hunk 5]
```diff
@@ -196,206 +196,52 @@ class AppWrapper extends React.Component {
...
   useEffect(() => {
-    document.title = isEnabled ? 'login1' : 'login2';
-  }, [isEnabled]);
+    document.title = decision.variationKey === 'login-new' ? 'login-new' : 'login-default';
+  }, [decision.enabled]);
 
   return (
     <p>
-      <a href={isEnabled ? '/login' : '/login2'}>{variables.loginText}</a>
+      <a href={decision.variationKey === 'login-new' ? '/login-new' : '/login-default'}>Click to login</a>
```
Comment (yavorona) [inline README.md:244]: shouldn't this be rendered based on `decision['enabled']` condition?

**Review by yavorona — CHANGES_REQUESTED** [review]
Looks great except for a few minor comments

**Review by yavorona — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -24,44 +24,33 @@ The React SDK is compatible with `React 16.3.0 +`
 import {
   createInstance,
   OptimizelyProvider,
-  OptimizelyExperiment,
```
Comment (fscelliott) [inline README.md:27]: ^^ not in scope of your changes but you might want to update in line 15:
'library of react components'
to 'library of react components and hooks to...'

File: README.md [hunk 7]
```diff
@@ -24,44 +24,33 @@ The React SDK is compatible with `React 16.3.0 +`
...
+const optimizelyClient = createInstance({
   sdkKey: 'your-optimizely-sdk-key',
 });
 
+function MyComponent() {
+  const [decision] = useDecision('sort-algorithm');
+  return (
+    <React.Fragment>
+      <SearchComponent algorithm={decision.variables.algorithm} />
+      { decision.variationKey === 'bluebutton' && <BlueButton /> }
```
Comment (fscelliott) [inline README.md:39]: I don't get how a blue vs green button relates to a 'sort-algorithm' flag.  Could variations be more like "relevant_first" and "popular_first" or "recommended_first" or something?

File: README.md [hunk 8]
```diff
@@ -108,13 +97,13 @@ const optimizely = createInstance({
 
 ## `<OptimizelyProvider>`
```
Comment (fscelliott) [inline README.md:99]: out of the scope of your changes, but if you're using the Decide API, you should also link to the latest docs, for which you have to specify v4.0 in the URL (until we make v4.0 the default).
so these links should be updated to v4.0:

so any links with
https://docs.developers.optimizely.com/full-stack/docs/...
should update to
https://docs.developers.optimizely.com/full-stack/v4.0/docs/

File: README.md [hunk 9]
```diff
@@ -130,14 +119,14 @@ Synchronous loading is the preferred method to ensure that Optimizely is always
 ```jsx
 import { OptimizelyProvider, createInstance } from '@optimizely/react-sdk';
 
-const optimizely = createInstance({
+const optimizelyClient = createInstance({
   datafile: window.datafile,
```
Comment (fscelliott) [inline README.md:123]: It's interesting we say that initializing from datafile is the 'preferred' method. I guess I would have thought that if synchronous init is important, the preferred method would be to init w/ both datafile & SDK key? docs state:

"When initializing with both the SDK key and datafile, the SDK will use the given datafile to start, then download the latest version of the datafile in the background"

**Pull request merged**